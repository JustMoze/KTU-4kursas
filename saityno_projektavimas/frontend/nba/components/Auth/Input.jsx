import React, { useState } from 'react'
import TextField from '@material-ui/core/TextField'
import styled from 'styled-components'

function Input({ handleChange, error, message, name, ...props }) {
  const WhiteBorderTextField = styled(TextField)`
    & label.Mui-focused {
      color: ${error ? 'red' : 'green'};
    }
    & label.Mui {
      color: ${error ? 'red' : 'green'};
    }
    & .MuiOutlinedInput-root {
      &.Mui-focused fieldset {
        border-color: ${error ? 'red' : 'green'};
      },
    }
    & .MuiOutlinedInput-root {
      & fieldset {
        border-color: ${error ? 'red' : 'green'};
      },
    }

    & p.MuiFormHelperText-root {
            color: ${error ? 'red' : '#323232'}
        }
    }
  `
  const ConfigureName = (name) => {
    return name.charAt(0).toUpperCase() + name.slice(1)
  }
  return (
    <WhiteBorderTextField
      error={error}
      label={ConfigureName(name)}
      helperText={error && message}
      variant="outlined"
      onChange={(text) => handleChange(text)}
      name={name}
      style={{
        width: '100%',
        marginBottom: 10,
      }}
      {...props}
    />
  )
}

export default Input
