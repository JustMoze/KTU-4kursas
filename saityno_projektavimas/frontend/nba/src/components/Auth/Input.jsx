import React from 'react'
import TextField from '@material-ui/core/TextField'
import styled from 'styled-components'

export const WhiteBorderTextField = styled(TextField)`
  & label.Mui-focused {
    color: ${(props) => (props.error ? 'red' : props.color)};
  }
  & label.Mui {
    color: ${(props) => (props.error ? 'red' : props.color)};
  }
  & .MuiOutlinedInput-root {
    &.Mui-focused fieldset {
      border-color: ${(props) => (props.error ? 'red' : props.color)};
    },
  }
  & .MuiOutlinedInput-root {
    & fieldset {
      border-color: ${(props) => (props.error ? 'red' : props.color)};
    },
  }

  & p.MuiFormHelperText-root {
          color: ${(props) => (props.error ? 'red' : '#323232')}
      }
  }
`
function Input({ handleChange, error, message, name, color, ...props }) {
  const ConfigureName = (name) => {
    name = name.slice(name.indexOf('_') + 1, name.length)
    return name.charAt(0).toUpperCase() + name.slice(1)
  }
  return (
    <WhiteBorderTextField
      color={color}
      error={error}
      label={ConfigureName(name)}
      helperText={error && message}
      variant="outlined"
      onChange={(text) => handleChange(text)}
      name={name}
      style={{
        width: '100%',
        marginBottom: 10,
      }}
      {...props}
    />
  )
}

export default Input
