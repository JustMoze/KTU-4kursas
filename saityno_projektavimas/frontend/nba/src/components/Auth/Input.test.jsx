import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import Input, {WhiteBorderTextField} from './Input'
import renderer from 'react-test-renderer';
import 'jest-styled-components'

configure({ adapter: new Adapter() })

describe('Input', () => {
    const handleChange = jest.fn();
    const initProps = {
      color: 'primary',
      error: false,
      message: '',
      name: 'login_email',
      value: 'kompotas@gmail.com',
      handleChange: handleChange,
      type: 'login'
    }
    it('Renders input correctly', () => {
      const customWrapper = shallow(<Input {...initProps} />)
      expect(toJson(customWrapper)).toMatchSnapshot();
    })

    const wrapper = shallow(<Input {...initProps} />)
    test('WhiteBorderTextField works', () => {
      const playerRow = renderer.create(<WhiteBorderTextField color={initProps.color} error={initProps.error} />).toJSON()
      expect(playerRow).toHaveStyleRule('color', initProps.color, {
          modifier: '& label.Mui-focused',
      })
      expect(playerRow).toHaveStyleRule('color', initProps.color, {
          modifier: '& label.Mui',
      })
      expect(playerRow).toHaveStyleRule('color', '#323232', {
          modifier: 'p.MuiFormHelperText-root',
      })
    })

    it('Updates the state', () => {
      const input = wrapper.find(WhiteBorderTextField);
      console.log('input', input.debug())
      input.simulate('change', {target: {text: 'kompotas1@gmail.com'}})
      console.log('input', input.debug())
    })
   
  })