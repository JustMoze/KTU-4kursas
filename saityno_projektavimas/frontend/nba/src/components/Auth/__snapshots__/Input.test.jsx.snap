// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Renders input correctly 1`] = `
<Input__WhiteBorderTextField
  color="primary"
  error={false}
  helperText={false}
  label="Email"
  name="login_email"
  onChange={[Function]}
  style={
    Object {
      "marginBottom": 10,
      "width": "100%",
    }
  }
  type="login"
  value="kompotas@gmail.com"
  variant="outlined"
/>
`;
